### ISSUE TITLE
Update highlight boxes in individual modules

### ISSUE THREAD

**Issue by drelliche** [arcus/education_modules#242]
Now that the new highlight boxes symbols have been incorporated, each module will need to be checked and updated.

- Every highlight box (except for answer boxes) should have a title in the accent color as is now shown in the example. These must be included as a script that can be copied directly from the example in the `a_sample_module` examples.
- There are now more options for highlight boxes, so make sure that the correct type is used for each box (i.e. a former `learn more` box might now be a `learn-more` or a `history` or a `cool-fact` or a `behind-the-scenes` box)
- ~~Update the module_template_version to `3.0.0`~~
- Update the version. This is a [**minor** update](https://github.com/arcus/education_modules/blob/main/versioning_guidelines.md#xminorx): x.MINOR.x

We can keep track of which modules have been updated here:

- [x] bash_103_combining_commands
- [x] bash_command_line_101
- [x] bash_command_line_102
- [x] bash_conditionals_loops
- [x] data_storage_models
- [x] data_visualization_in_ggplot2
- [x] #427
- [x] #428
- [x] demystifying_geospatial_data
- [x] demystifying_python
- [x] demystifying_sql
- [x] directories_and_file_paths
- [x] git_creation_and_tracking
- [x] git_history_of_project
- [x] git_intro
- [x] git_setup_mac_and_linux
- [x] git_setup_windows
- [x] how_to_troubleshoot
- [x] learning_to_learn
- [x] omics_orientation
- [x] pandas_transformation
- [x] python_basics_writing_python_code
- [x] r_basics_introduction
- [x] r_basics_transform_data
- [x] r_basics_visualizae_data
- [x] r_missing_values
- [x] #429
- [x] reproducibility
- [x] sql_basic
- [x] #437
- [x] statistical_tests
- [x] tidy_data
- [x] using_redcap_api

**Comment by franzenr** [comment]
Not sure if we're hoping to push this through soon, but if it ends up being the case that it seems like I'll be able to get an action running that yells at us whenever we try to commit a module update w/o first incrementing the version number (I haven't had a chance to play with this in a little while bc it's been deprioritized temporarily in favor of getting our recruitment/study data collection up and running in REDCap. I can try to find time for a brief feasibility/timeline estimate sooner, though) are we open to kicking the can down the road on this until I'm able to get that implemented?

We could always have folks going through their modules to make updates as there is time but just not pushing any changes to main until it's running. I'm just a little worried at the potential for something of this scale to result in a LOT of version #s not being incremented appropriately bc it's super easy to forget. 

Getting these up to date may also be a good "background"/"filler" task to have for our January interns so they have something to always return to if they hit a wall on other tasks until we're able to get them up and running again.

**Comment by franzenr** [comment]
I'm also not sure how we want to approach the module template number within the module.... 

This is something I've had swimming in the back of my mind for a while, but until now there hasn't been impetus to pull it out. 

Part of the purpose of recording the module template number is to allow us to identify where each module stands in adherence to our standards, especially for things we can't automate the QA of.
As an example, I know Reproducibility module has a pretty old template number (in fact, it may have even been written before we started adding template versions). If in the process of checking the highlight boxes, we update that template number to 3.0.0 (side note, this is what the new template version number should actually be, see my comment on #241 ) then we would mistakenly think that Reproducibility meets all of the other template updates made since it was last edited, when in reality, the only new criteria it meets is the highlight boxes, and everything else is still out of date. 

While this is a clear problem, the solution is much less clear to me. One thing that may be helpful is to implement some sort of semantic change log where we record comments for each version number. That way, whenever each module is updated in ways relating to the template, we can go through and ensure that it meets each of the specified criteria for every single template update that has occurred during all versions between its current version and our newest template version. The difficulty of this is ensuring that we always remember to add to this, and that when we add to it, we don't leave anything out and include sufficient detail to allow us to be confident we are meeting all the criteria. 

Another approach is to force us to immediately update all modules to meet the new template version each time we change the template. This will likely be very annoying at times, BUT it also seems like the most foolproof way to avoid downstream collisions, incompatibilities, or inconsistent module qualities. On one hand, this approach feels less anxiety inducing to me because I'm always anticipating all the ways we can mess things up, but I am also very sympathetic to the resentment we will may develop for this self-enforced rule over time (myself included). 

I'm sure there are also middle-of-the-road approaches and other possibilities, and I'm happy to hear what those are! 

We can also move this discussion elsewhere (maybe a separate issue?) if that seems more suitable.

**Comment by pm0kjp** [comment]
Finally did mine, waiting on PR but I'm going to close anyway!
